EMaligner/jsongz.py

```diff
@@ -0,0 +1,43 @@
+import json
+import gzip
+import os
+
+
+def _convert_ext(filepath, compress):
+    b, e = os.path.splitext(filepath)
+    filepath = b + '.json'
+    if compress:
+        filepath += '.gz'
+    return filepath
+
+
+def _check_ext(filepath):
+    compress = False
+    b, e = os.path.splitext(filepath)
+    if e == '.gz':
+        compress = True
+    return compress
+
+
+def dump(obj, filepath, compress=None, encoding='utf-8', *args, **kwargs):
+    if not compress:
+        compress = _check_ext(filepath)
+    filepath = _convert_ext(filepath, compress)
+    if compress:
+        with gzip.GzipFile(filepath, 'w') as f:
+            f.write(json.dumps(obj, *args, **kwargs).encode(encoding))
+    else:
+        with open(filepath, 'w') as f:
```

Comment:
If encoding does not apply to json files it should be documented.

Reply:
added docs and made encoding apply to json files also.